overlord/ifacestate/ifacestate_test.go

```diff
@@ -1740,6 +1740,63 @@ slots:
 	_ = s.getConnection(c, "consumer", "plug", "producer", "slot")
 }
 
+func (s *interfaceManagerSuite) TestForgetUndo(c *C) {
+	s.mockIfaces(c, &ifacetest.TestInterface{InterfaceName: "test"}, &ifacetest.TestInterface{InterfaceName: "test2"})
+
+	s.mockSnap(c, consumerYaml)
+	s.mockSnap(c, producerYaml)
+
+	// plug3 and slot3 do not exist, so the connection is not in the repository.
+	connState := map[string]interface{}{
+		"consumer:plug producer:slot":   map[string]interface{}{"interface": "test"},
+		"consumer:plug3 producer:slot3": map[string]interface{}{"interface": "test2"},
+	}
+
+	s.state.Lock()
+	s.state.Set("conns", connState)
+	s.state.Unlock()
+
+	// Initialize the manager. This registers both snaps and reloads the connection.
+	mgr := s.manager(c)
+
+	// sanity
+	_ = s.getConnection(c, "consumer", "plug", "producer", "slot")
+
+	s.state.Lock()
+	change := s.state.NewChange("disconnect", "...")
+	ts, err := ifacestate.Forget(s.state, mgr.Repository(), &interfaces.ConnRef{
+		PlugRef: interfaces.PlugRef{Snap: "consumer", Name: "plug3"},
+		SlotRef: interfaces.SlotRef{Snap: "producer", Name: "slot3"}})
+	c.Assert(err, IsNil)
+	// inactive connection, only the disconnect task (no hooks)
+	c.Assert(ts.Tasks(), HasLen, 1)
+	task := ts.Tasks()[0]
+	c.Check(task.Kind(), Equals, "disconnect")
+	var forgetFlag bool
+	c.Assert(task.Get("forget", &forgetFlag), IsNil)
+	c.Check(forgetFlag, Equals, true)
+
+	change.AddAll(ts)
+	terr := s.state.NewTask("error-trigger", "provoking total undo")
+	terr.WaitAll(ts)
+	change.AddTask(terr)
+
+	// Run the disconnect task and let it finish.
+	s.state.Unlock()
+	s.settle(c)
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	// Ensure that disconnect task was undone
+	c.Assert(task.Status(), Equals, state.UndoneStatus)
+
+	var conns map[string]interface{}
+	c.Assert(s.state.Get("conns", &conns), IsNil)
+	c.Assert(conns, DeepEquals, connState)
+
+	_ = s.getConnection(c, "consumer", "plug", "producer", "slot")
```

Comment:
super minor nitpick, but couldn't this just be 

```suggestion
	s.getConnection(c, "consumer", "plug", "producer", "slot")
```
?